modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkOperationModelParser.java

```diff
@@ -403,14 +403,14 @@ private Stream<ComponentAstWithHierarchy> getOperationsAstRecursiveStream(Map<St
 
           if (filtered.contains(componentAst) || ignoreCondition.test(componentAstWithHierarchy)) {
             return Stream.empty();
+          } else if (filterCondition.test(componentAstWithHierarchy)) {
```

Comment:
why this change? isn't this altering behavior? With the prior logic, it'd be possible for a component to be returned even though the filtered condition would also be true

Reply:
The problem is that before a scope/router that should have been ignored won't be ignored. Since in the next line we check if there is an operation model, and then we proceed... but now the scopes and routers are operations, so we find an operation model for them, and we don't ignore them. 
The example is for checking if an operation is transactional, and it has an async which inside has a transactional try. That operation is not transactional, and we have this into consideration in the ignore corresponding to isTransactional.. but since now there is an operation model, we don't check if it has to be ignored

Reply:
And that logic was added in particular for cases like this one :/